src/functions/getDelegatedBooking/framework/__tests__/handler.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { Mock, It, Times } from 'typemoq';
+import { APIGatewayEvent, Context } from 'aws-lambda';
+const lambdaTestUtils = require('aws-lambda-test-utils');
+
+import { handler } from '../handler';
+import * as createResponse from '../../../../common/application/utils/createResponse';
+import * as FindDelegatedBooking from '../../../../common/application/delegated-booking/FindDelegatedBooking';
+import { DelegatedBookingNotFoundError } from '../../../../common/domain/errors/delegated-booking-not-found-error';
+import { tokens } from '../__mocks__/authentication-token.mock';
+import { booking } from '../../../../common/application/service/__mocks__/test-delegated-booking-decompressor';
+
+describe('getDelegatedBooking handler', () => {
+  let dummyApigwEvent: APIGatewayEvent;
+  let dummyContext: Context;
+  let createResponseSpy: jasmine.Spy;
+
+  const moqFindDelegatedBooking = Mock.ofInstance(FindDelegatedBooking.findDelegatedBooking);
+
+  beforeEach(() => {
+    moqFindDelegatedBooking.reset();
+
+    createResponseSpy = spyOn(createResponse, 'default');
+    dummyApigwEvent = lambdaTestUtils.mockEventCreator.createAPIGatewayEvent({
+      pathParameters: {
+        applicationReference: '12345678910',
+      },
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: tokens.employeeId_12345678,
+      },
+    });
+    dummyApigwEvent.requestContext.authorizer = { examinerRole: 'DLG' };
+    dummyContext = lambdaTestUtils.mockContextCreator(() => null);
+    spyOn(FindDelegatedBooking, 'findDelegatedBooking').and.callFake(moqFindDelegatedBooking.object);
+  });
+
+  describe('given the FindJournal returns a journal', () => {
+    it('should return a successful response with the journal', async () => {
+      moqFindDelegatedBooking.setup(x => x(It.isAny())).returns(() => Promise.resolve(booking));
+      createResponseSpy.and.returnValue({ statusCode: 200 });
+
+      const resp = await handler(dummyApigwEvent, dummyContext);
+
+      expect(resp.statusCode).toBe(200);
+      expect(createResponse.default).toHaveBeenCalledWith(booking);
+    });
+  });
+
+  describe('given FindJournal throws a JournalNotFound error', () => {
+    it('should return HTTP 404 NOT_FOUND', async () => {
+      moqFindDelegatedBooking.setup(x => x(It.isAny())).throws(new DelegatedBookingNotFoundError());
+      createResponseSpy.and.returnValue({ statusCode: 404 });
```

Comment:
What's the need to mock `createResponse()`? The tests should pass using the real createResponse function and it would make these more reliable tests

Reply:
Fair point, I will remove the createResponseSpy to show the statusCode being returned correctly.